[LIU SIYI] iP

64-1 opened the pull request.

64-1 committed:
change name of Duke.java into Erii.java

64-1 committed:
Added UserDetail and TaskManager classes

JingHaoooo commented on src/main/java/UserDetails.java:
```diff
@@ -0,0 +1,88 @@
+        while (true) {
+            String input = scanner.nextLine().trim();
+
+            switch (input) {
+                case "1":
+                    gender = "Male";
+                    break;
+                case "2":
+                    gender = "Female";
+                    break;
+                case "3":
+                    gender = "Other";
+                    break;
+                default:
+                    System.out.println("Invalid option selected. Please enter 1, 2, or 3.");
+                    continue;
+            }
+            break; // Exit the loop once a valid input is received.
```
There should not be any indentation for case clauses to follow the coding standards.

JingHaoooo commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,30 @@
+        String gender = UserDetails.getGenderSelection();
+
+
+        String message = String.format(
+                "Verification passed.\nOptions enabled.\n%s Born on %s\n%s\nID A.D.0013\n" +
+                        "Rank 'S'\nListed in the Kassel Academy roster.\n" +
+                        "Database access granted\nAccount activated\nCourse schedule generated\n" +
+                        "I am Erii, the secretary of Kassel Academy, pleased to serve you.",
+                name, birthday, gender);
```
I like the line wrapping. It makes it more readable.

JingHaoooo commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,39 @@
+// TaskManager.java
+public class TaskManager {
+    private static final String[] tasks = new String[100];
```
Can consider setting 100 to a constant variable.

CXIA17 replied:
Try to make Task as a class so it can be more manageable in the future.

JingHaoooo commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,39 @@
+    public static void addTask(String task) {
+        if (taskCount < tasks.length) {
+            tasks[taskCount++] = task;
+            System.out.println("____________________________________________________________");
```
Can consider changing divider to a constant.

CXIA17 commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,27 @@
-| | | | | | | |/ / _ \
-| |_| | |_| |   <  __/
-|____/ \__,_|_|\_\___|
+
+          _____                    _____                    _____                    _____          
+         /\    \                  /\    \                  /\    \                  /\    \         
+        /::\    \                /::\    \                /::\    \                /::\    \        
+       /::::\    \              /::::\    \               \:::\    \               \:::\    \       
+      /::::::\    \            /::::::\    \               \:::\    \               \:::\    \      
+     /:::/\:::\    \          /:::/\:::\    \               \:::\    \               \:::\    \     
+    /:::/__\:::\    \        /:::/__\:::\    \               \:::\    \               \:::\    \    
+   /::::\   \:::\    \      /::::\   \:::\    \              /::::\    \              /::::\    \   
+  /::::::\   \:::\    \    /::::::\   \:::\    \    ____    /::::::\    \    ____    /::::::\    \  
+ /:::/\:::\   \:::\    \  /:::/\:::\   \:::\____\  /\   \  /:::/\:::\    \  /\   \  /:::/\:::\    \ 
+/:::/__\:::\   \:::\____\/:::/  \:::\   \:::|    |/::\   \/:::/  \:::\____\/::\   \/:::/  \:::\____\
+\:::\   \:::\   \::/    /\::/   |::::\  /:::|____|\:::\  /:::/    \::/    /\:::\  /:::/    \::/    /
+ \:::\   \:::\   \/____/  \/____|:::::\/:::/    /  \:::\/:::/    / \/____/  \:::\/:::/    / \/____/ 
+  \:::\   \:::\    \            |:::::::::/    /    \::::::/    /            \::::::/    /          
+   \:::\   \:::\____\           |::|\::::/    /      \::::/____/              \::::/____/           
+    \:::\   \::/    /           |::| \::/____/        \:::\    \               \:::\    \           
+     \:::\   \/____/            |::|  ~|               \:::\    \               \:::\    \          
+      \:::\    \                |::|   |                \:::\    \               \:::\    \         
+       \:::\____\               \::|   |                 \:::\____\               \:::\____\        
+        \::/    /                \:|   |                  \::/    /                \::/    /        
+         \/____/                  \|___|                   \/____/                  \/____/         
+                                                                                                    
+                                                                                                 
```
A cool name, very aesthetically pleasing

CXIA17 commented on src/main/java/UserDetails.java:
```diff
@@ -0,0 +1,88 @@
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
```
A very interesting feature, maybe try to breakuser information down to seperate classes in order to make it more Object-Oriented

JingHaoooo commented on src/main/java/UserDetails.java:
```diff
@@ -0,0 +1,88 @@
+     *
+     * @return the user's full name
+     */
+    public static String getName() {
```
Naming of variables are good as it follows the naming conventions of using camelCase.

JingHaoooo reviewed (commented):
Overall it looks good. Although there were some minor coding standard violations.

CXIA17 commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,39 @@
+        System.out.println("____________________________________________________________");
+    }
+
+    public static boolean processCommand(String command) {
+        if ("list".equalsIgnoreCase(command)) {
+            listTasks();
+            return true;
+        } else if ("bye".equalsIgnoreCase(command)) {
+            System.out.println("____________________________________________________________");
+            System.out.println(" Bye. Hope to see you again soon!");
+            System.out.println("____________________________________________________________");
+            return false;
+        } else {
+            addTask(command);
+            return true;
+        }
```
Good indentation

CXIA17 commented on src/main/java/UserDetails.java:
```diff
@@ -0,0 +1,88 @@
+        while (true) {
+            String input = scanner.nextLine().trim();
+
+            switch (input) {
+                case "1":
+                    gender = "Male";
+                    break;
+                case "2":
+                    gender = "Female";
+                    break;
+                case "3":
+                    gender = "Other";
+                    break;
+                default:
+                    System.out.println("Invalid option selected. Please enter 1, 2, or 3.");
+                    continue;
```
case and switch should be aligned, invalid indentation

hafizuddin-a commented on src/main/java/Erii.java:
```diff
@@ -0,0 +1,27 @@
+public class Erii {
+    public static void main(String[] args) {
+        String art = 
```
perhaps move this to another file?

CXIA17 replied:
Make variable names for constant all uppercase.

hafizuddin-a commented on src/main/java/UserDetails.java:
```diff
@@ -0,0 +1,88 @@
+import java.util.Scanner;
+import java.util.regex.Matcher;
```
i like that you use regex

CXIA17 replied:
Import classes listed explicitly, good

hafizuddin-a commented on src/main/java/UserDetails.java:
```diff
@@ -0,0 +1,88 @@
+        System.out.println("Please enter your full name (First Name Last Name): ");
+        String name;
+        while (true) {
+            name = scanner.nextLine().trim();
```
good using trim to remove unnecessary characters

hafizuddin-a commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    public static boolean processCommand(String command) {
+        if ("list".equalsIgnoreCase(command)) {
```
good that you have a method to ignore case

hafizuddin-a reviewed (commented):
good code quality

CXIA17 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,30 @@
+
+public class Main {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        String name = UserDetails.getName();
+        String birthday = UserDetails.getBirthday();
+        String gender = UserDetails.getGenderSelection();
```
Good, very specific names given

64-1 committed:
Added ControlPanel class and modified taskmanager

64-1 committed:
Optimised ControlPanel and TaskManager

64-1 committed:
Encapsulated task priority validation

64-1 committed:
Added displayMenu method for ControlPanel

64-1 committed:
Adjusted validatePriority method

64-1 committed:
Merge branch 'branch-Level-5'

64-1 committed:
Refactor project structure into packages

Organized classes into dedicated packages for better modularity and maintainability. Created com.erii as the base package and subdivided functionalities into core, ui, and user packages respectively. Addresses issue #123 for project structure optimization.

64-1 committed:
Update README.md

64-1 committed:
Update README.md

64-1 committed:
Update README.md

64-1 committed:
Update README.md

64-1 committed:
Update README.md

64-1 committed:
Update README.md

64-1 committed:
Update README.md

64-1 committed:
Add delete task function

sevenseasofbri commented on src/main/java/com/erii/Main.java:
```diff
@@ -0,0 +1,27 @@
+
+public class Main {
+    public static void main(String[] args) {
+        // Prompt the user to enter their details
+        String name = UserDetails.getName();
+        String birthday = UserDetails.getBirthday();
+        String gender = UserDetails.getGenderSelection();
+
+
+        // Print the user details with a welcome message
+        String message = String.format(
+                "Verification passed.\nOptions enabled.\n%s Born on %s\n%s\nID A.D.0013\n" +
+                        "Rank 'S'\nListed in the Kassel Academy roster.\n" +
+                        "Database access granted\nAccount activated\nCourse schedule generated\n" +
+                        "I am Erii, the secretary of Kassel Academy, pleased to serve you.",
+                name, birthday, gender);
+
+        System.out.println(message);
+
+        //Call control panel to start the task manager
+        ControlPanel.main(args);
```
These comments are not required as it is quite clear from the code already.

sevenseasofbri commented on src/main/java/com/erii/core/Erii.java:
```diff
@@ -0,0 +1,29 @@
+
+public class Erii {
+    public static void main(String[] args) {
+        String art = 
```
consider storing this `art` as a static final variable

sevenseasofbri commented on src/main/java/com/erii/core/TaskManager.java:
```diff
@@ -0,0 +1,191 @@
+
+public class TaskManager {
+
+    public enum Priority {
```
good use of enum

sevenseasofbri commented on src/main/java/com/erii/core/TaskManager.java:
```diff
@@ -0,0 +1,191 @@
+        SS, S, A, B, C, D
+    }
+
+    public abstract class Task {
```
avoid declaring multiple classes within a same file, try an separate them into differen files (to improve readability)

sevenseasofbri commented on src/main/java/com/erii/core/TaskManager.java:
```diff
@@ -0,0 +1,191 @@
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + task);
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+        System.out.println("____________________________________________________________");
```
you can store is line as a static final variable instead

sevenseasofbri commented on src/main/java/com/erii/core/TaskManager.java:
```diff
@@ -0,0 +1,191 @@
+        protected String by;
+
+        public Deadline(String name, String description, String by, Priority priority) {
+            super(name, description, priority); // Fixed: Correctly initialize the parent class
```
avoid writing comments as reminders and todos for yourself

sevenseasofbri commented on src/main/java/com/erii/core/TaskManager.java:
```diff
@@ -0,0 +1,191 @@
+        System.out.println("Tasks sorted by type.");
+    }
+
+    // public void sortListByDeadline() {
+    //     tasks.sort((Task t1, Task t2) -> {
+    //         if (t1 instanceof Deadline && t2 instanceof Deadline) {
+    //             return ((Deadline) t1).by.compareTo(((Deadline) t2).by);
+    //         } else {
+    //             return 0;
+    //         }
+    //     });
+    //     System.out.println("Tasks sorted by deadline.");
+    // }
```
remove commented code, it reduces redability

sevenseasofbri commented on src/main/java/com/erii/ui/ControlPanel.java:
```diff
@@ -0,0 +1,153 @@
+        // Splitting the input at " /" to separate description and priority
+        String[] parts = input.split(" ?/ ?");
+        if (parts.length < 2 || parts[1].isEmpty()) {
+            System.out.println("\"Incorrect format. Please ensure the task description is followed by '/' and a priority value (e.g., 'slain a dragon /SS').\"");
```
Line length should be no longer than 120 chars. line contains 149 characters, consider splitting it into two lines 

https://se-education.org/guides/conventions/java/basic.html#layout

sevenseasofbri commented on src/main/java/com/erii/ui/ControlPanel.java:
```diff
@@ -0,0 +1,153 @@
+        TaskManager.Priority priority;
+        try{
+            priority = TaskManager.Priority.valueOf(parts[1].trim().toUpperCase());
+        }catch(IllegalArgumentException e){
```
spacing between } and catch required

sevenseasofbri commented on src/main/java/com/erii/ui/ControlPanel.java:
```diff
@@ -0,0 +1,153 @@
+    private static void addDeadlineTask(String input, TaskManager taskManager) {
+        String[] parts = input.split(" ?/by |  ?/ ?");
+        if(parts.length < 3 || parts[1].isEmpty() || parts[2].isEmpty()){
+            System.out.println("Incorrect format. Please ensure the task description is followed by '/by' and a deadline date, and then a priority value (e.g., 'submit report /by 2021-09-30 /SS').");
```
similarly here, split into 2 lines to avoid exceeding the line character limit.

sevenseasofbri commented on src/main/java/com/erii/ui/ControlPanel.java:
```diff
@@ -0,0 +1,153 @@
+        }
+    }
+
+    private static String getAddress() {
```
the name of this function is a little misleading perhaps? I understand it probably refers to how to 'address' someone, but maybe use a word like `salutation` or `userTitle` instead?

sevenseasofbri commented on src/main/java/com/erii/user/UserDetails.java:
```diff
@@ -0,0 +1,105 @@
+ * The UserDetails class provides methods to retrieve user details such as name, birthday, and gender.
+ */
+public class UserDetails {
+    private static final Scanner scanner = new Scanner(System.in);
+    private static final Pattern datePattern = Pattern.compile("^\\d{2}/\\d{2}/\\d{4}$");
```
Constant names must be all uppercase using underscore to separate words (aka SCREAMING_SNAKE_CASE).

https://se-education.org/guides/conventions/java/basic.html#naming

sevenseasofbri commented on src/main/java/com/erii/user/UserDetails.java:
```diff
@@ -0,0 +1,105 @@
+public class UserDetails {
+    private static final Scanner scanner = new Scanner(System.in);
+    private static final Pattern datePattern = Pattern.compile("^\\d{2}/\\d{2}/\\d{4}$");
+    private static String UserName = "";
+    private static String UserBirthday = "";
+    private static String UserGender = "";
```
Variable names must be in camelCase.

https://se-education.org/guides/conventions/java/basic.html#naming

sevenseasofbri commented on src/main/java/com/erii/ui/ControlPanel.java:
```diff
@@ -0,0 +1,153 @@
+
+public class ControlPanel {
+
+    public static void main(String[] args) {
```
this method is really long (>30 LOC) shorten it by abstracting out parts of code into functions 

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods

sevenseasofbri reviewed (commented):
Good job, you can improve upon abstraction of the code. Else, well done. 👍🏽

64-1 committed:
Add data saving function and encapsulate control panel class

64-1 committed:
Adjust data storage class

64-1 committed:
After braching gradle

64-1 committed:
Revert "After braching gradle"

This reverts commit 5eee43ef5dc73f34e72a39a25600fb6461a14895.

64-1 committed:
Creater JAR file

64-1 committed:
Update annotation

64-1 committed:
Update README.md

64-1 committed:
Update README.md

64-1 committed:
Update README.md

64-1 committed:
Enhance event and deadline date handling with time support

64-1 committed:
Merge pull request #1 from 64-1/branch-Level-8

Enhance event and deadline date handling with time support

64-1 committed:
Implement task search by keyword

64-1 committed:
Merge pull request #2 from 64-1/branch-Level-9

Implement task search by keyword

64-1 committed:
Add documentation

64-1 committed:
Add documentation

(cherry picked from commit 007e4ab0d5e6d2be803b583b9c499465f877a1b6)

64-1 committed:
Revert "Add documentation"

This reverts commit 007e4ab0d5e6d2be803b583b9c499465f877a1b6.

64-1 committed:
Merge branch 'branch-Level-9'

64-1 committed:
Update README.md

64-1 committed:
Update README.md

64-1 committed:
Update README.md

64-1 committed:
Update README.md

64-1 committed:
Update README.md

64-1 committed:
Update README.md

64-1 committed:
no message

64-1 committed:
Update Erii Functional Overview

64-1 committed:
Delete ip.jar

64-1 committed:
Update README.md with user guide and main features

64-1 committed:
Refactor data storage and control panel classes

64-1 committed:
Remove tasks and user details

64-1 committed:
Adjust user details and task management functionality

64-1 committed:
Add taskToFileString() and fileStringToTask() methods

64-1 committed:
Refactor taskToFileString and fileStringToTask methods

64-1 committed:
Refactor TaskManager class by removing empty lines at the beginning and end of the file

64-1 committed:
Add new options for listing tasks by date and searching

64-1 committed:
Update README.md

64-1 committed:
Refactor task type validation and add search by date feature

64-1 committed:
Merge branch 'master' of https://github.com/64-1/ip

64-1 committed:
Update userDetails.txt and EXPECTED.TXT

64-1 committed:
Update .gitignore to include Erii.jar
